csharp/private/tools/assembly-runner/main.cc

```diff
@@ -0,0 +1,49 @@
+#include <iostream>
+#include <unistd.h>
+#include "tools/cpp/runfiles/runfiles.h"
+
+using bazel::tools::cpp::runfiles::Runfiles;
+
+int main(int argc, char** argv) {
+  std::string error;
+
+  auto runfiles = Runfiles::Create(argv[0], &error);
+
+  if (runfiles == nullptr) {
+    std::cerr << "Couldn't load runfiles: " << error << std::endl;
+    return 1;
+  }
+
+  auto dotnet = runfiles->Rlocation("{DotnetExe}");
+
+  if (dotnet.empty() || access(dotnet.c_str(), F_OK) == -1) {
+    std::cerr << "Couldn't find the .NET runtime" << std::endl;
+    return 1;
+  }
+
+  auto exe = runfiles->Rlocation("{TargetExe}");
+
+  if (exe.empty() || access(dotnet.c_str(), F_OK) == -1) {
+    std::cerr << "Couldn't find the target .exe" << std::endl;
+    return 1;
+  }
+
+  auto dotnet_argv = new char*[argc + 4];
+
+  // dotnet wants this to either be dotnet or dotnet.exe but doesn't have a
+  // preference otherwise.
+  dotnet_argv[0] = (char*)"dotnet";
+  dotnet_argv[1] = (char*)"--runtimeconfig";
+  dotnet_argv[2] = (char*)"{RuntimeConfigJson}";
+
+  dotnet_argv[3] = (char*)exe.c_str();
```

Comment:
For .NET Framework things we probably don't want to wrap with `dotnet`? Not sure if that works.
The wrapping would look similar for #51 I think... I think we could share this template with some complexification.

Eventually for #9 and the assembly loading stuff we'll want to invoke a different C# exe and path this `exe.c_str()` to the first one. The first one will overload assembly resolution to use the runfiles manifest. So I think we will want some form of wrapping no matter what platform/framework, it will just change slightly depending on those things.